project/src/js/lichess/friends.js

```diff
@@ -1,8 +1,40 @@
-import { caseInsensitiveSort } from '../utils';
-
 var onlineFriends = [];
 
+var makeUser = function(name, isPlaying) {
+    return {
+      'name' : name,
+      'isPlaying' : isPlaying,
+    }
+};
+
+/** Compares usernames for equality, ignoring prefixed titles (such as GM) */
+var isSameUser = function(userId, name) {
+  var id = (name.indexOf(' ') >= 0) ? name.split(' ')[1] : name;
+  return id.toLowerCase() === userId;
+};
+
+var findByUsername = function(n) {
+  return onlineFriends.filter(function(u) {
+    return isSameUser(n.toLowerCase(), u.name);
+  })[0];
```

Comment:
I assume we don't care about browser compatibility issues with 'find' (https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/find) since this is a cordova app?